app/controllers/api/v0/articles_controller.rb

```diff
@@ -62,22 +63,29 @@ def create
 
       def update
         @article = Article.find(params[:id])
+        not_found if @article.user_id != @user.id && @user.has_role?(:super_admin)
```

Comment:
I think you forgot a `!` and meant `and does NOT have super admin role`?

```suggestion
        not_found if @article.user_id != @user.id && !@user.has_role?(:super_admin)
```